funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -347,42 +353,90 @@ def stop_endpoint(self, name):
                     self.logger.info("Endpoint <{}> is now stopped".format(self.name))
 
             except OSError:
-                self.logger.warning("Endpoint {} could not be terminated".format(self.name))
-                self.logger.warning("Attempting Endpoint {} cleanup".format(self.name))
+                self.logger.warning("Endpoint <{}> could not be terminated".format(self.name))
+                self.logger.warning("Attempting Endpoint <{}> cleanup".format(self.name))
                 os.remove(pid_file)
                 sys.exit(-1)
+        # The process is not active, but the PID file exists and needs to be deleted
+        elif pid_check['exists']:
+            os.remove(pid_file)
+            self.logger.info("Endpoint <{}> has been cleaned up.".format(self.name))
         else:
             self.logger.info("Endpoint <{}> is not active.".format(self.name))
 
     def delete_endpoint(self, name):
         self.name = name
         endpoint_dir = os.path.join(self.funcx_dir, self.name)
 
-        # If endpoint currently running, stop it.
-        pid_file = os.path.join(endpoint_dir, "daemon.pid")
-        active = os.path.exists(pid_file)
-        if active:
-            stop_endpoint(self.name)
+        if not os.path.exists(endpoint_dir):
+            self.logger.warning("Endpoint <{}> does not exist".format(self.name))
+            sys.exit(-1)
+
+        # stopping the endpoint should handle all of the process cleanup before
+        # deletion of the directory
+        self.stop_endpoint(self.name)
 
         shutil.rmtree(endpoint_dir)
+        self.logger.info("Endpoint <{}> has been deleted.".format(self.name))
 
-    def check_pidfile(self, filepath, match_name):
+    def check_pidfile(self, filepath, match_name='funcx-endpoint', log=False):
```

Comment:
Why do we need the `log` option? By default, we should be logging these because the loglines here are important

Reply:
The function is now being used in new places where it wasn't before, like `funcx-endpoint list`. I don't think logs about all the pidfiles should be shown for that command at least. But I agree that logging should be swapped to the default behavior.

Reply:
Also, `check_pidfile` logs give instructions about cleaning up the endpoint using `funcx-endpoint stop`, but now I have changed the implementation of `funcx-endpoint stop` such that it uses the method `check_pidfile` as well so I think it could be confusing if the `funcx-endpoint stop` command suggests using `funcx-endpoint stop` in its own logs.